utils/utils/dataset.py

```diff
@@ -8,53 +8,86 @@
 from torch.utils.data import Dataset
 from sat.helpers import print_rank0
 
-def find_all_files(path, suffix=".jpg"):
+# Define a function to find all target files
+def find_all_files(path, suffixes=(".jpg", ".png")):
     target_files = []
     for cur_dir, _, files in os.walk(path, followlinks=True):
         for f in files:
-            if f.endswith(suffix):
+            if f.endswith(suffixes):
                 target_files.append(os.path.join(cur_dir, f))
-    print_rank0(f'find {len(target_files)} files...')
+    print_rank0(f'Found {len(target_files)} files...')
     return target_files
 
 class ItemDataset(Dataset):
+    # Initialization function, set image processor, text processor, data directories, etc.
     def __init__(self, image_processor, text_processor, args, data_dirs, cross_image_processor=None, **kwargs):
         super().__init__()
         self.data = self.load_data(data_dirs)
         self.image_processor, self.text_processor, self.cross_image_processor = image_processor, text_processor, cross_image_processor
+        self.data_dirs = data_dirs
+        self.data = self.load_data()
+        print_rank0(f"Dataset initialized with {len(self.data)} samples.")
     
+    # Function to process images
     def process_img(self, img):
         img_dict = {'vision': self.image_processor(img)}
         if self.cross_image_processor:
             img_dict.update({'cross': self.cross_image_processor(img)})
+        print_rank0(f"Image processed.")
         return img_dict
     
+    # Function to process text
     def process_text(self, answer, prompt):
-        return self.text_processor(answer, prompt)
+        processed_text = self.text_processor(answer, prompt)
+        print_rank0(f"Text processed.")
+        return processed_text
     
-    def load_data(self, data_dir):
-        all_files = find_all_files(data_dir, suffix=".jpg")
-        print_rank0(f"find {len(all_files)} samples in all...")
-        return all_files
+    # Function to load data
+    def load_data(self, data_dirs):
+        image_dir = os.path.join(self.data_dirs, 'images')
+        label_dir = os.path.join(self.data_dirs, 'labels')
+        image_files = self.find_all_files(image_dir, suffixes=(".jpg", ".png"))
+        
+        # Check if label directory exists
+        if os.path.exists(label_dir):
+            label_files = self.find_all_files(label_dir, suffix=".json")
+        else:
+            # If label directory does not exist, use image file names as labels
+            label_files = [os.path.splitext(os.path.basename(f))[0] for f in image_files]
```

Comment:
You may need to sort label_files and image_files before you zip them as `find_all_files` doesn't guarantee the result have the same order.

Reply:
Please review the updated `load_data` function.